### PR TITLE
Create common component for EmailField, PasswordField

### PR DESCRIPTION
**Pull request by anastasiaShpakava**

### CODE REVIEW

File: app/src/main/kotlin/com/softteco/template/ui/components/PasswordField.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            ConditionRow(
+                condition = stringResource(R.string.registration_password_condition2),
+                check = isPasswordHasUpperCase
+            )
+        }
+    }
+}
+
+@Composable
+fun ConditionRow(
```
Comment (bstolyarov) [inline app/src/main/kotlin/com/softteco/template/ui/components/PasswordField.kt:102]: Could be private

File: app/src/main/kotlin/com/softteco/template/ui/components/PasswordField.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import com.softteco.template.R
+import com.softteco.template.ui.feature.PasswordFieldState
+import com.softteco.template.ui.theme.Dimens
+
+@Composable
+fun PasswordField(
+    passwordValue: String,
+    onPasswordChanged: (String) -> Unit,
+    fieldStatePassword: PasswordFieldState,
+    isPasswordHasMinimum: Boolean?,
```
Comment (bstolyarov) [inline app/src/main/kotlin/com/softteco/template/ui/components/PasswordField.kt:39]: It seems false could be used as the default value, then the nullable option could be removed.

**Review by bstolyarov — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/softteco/template/ui/components/PasswordField.kt (hunk 2 above)
Comment (anastasiaShpakava) [reply app/src/main/kotlin/com/softteco/template/ui/components/PasswordField.kt:39]: it was created to know, if  we should show validation fields or not:
```
    if (isPasswordHasMinimum != null && isPasswordHasUpperCase != null) {
        Spacer(modifier = Modifier.height(Dimens.PaddingDefault))
        Column(verticalArrangement = Arrangement.spacedBy(Dimens.PaddingDefault)) {
            ConditionRow(
                condition = stringResource(R.string.registration_password_condition1),
                check = isPasswordHasMinimum
            )
            ConditionRow(
                condition = stringResource(R.string.registration_password_condition2),
                check = isPasswordHasUpperCase
            )
        }
    }
```

File: app/src/main/kotlin/com/softteco/template/ui/components/EmailField.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import androidx.compose.ui.res.stringResource
+import com.softteco.template.R
+import com.softteco.template.ui.feature.EmailFieldState
+
+@Composable
+fun EmailField(
+    emailValue: String,
+    onEmailChanged: (String) -> Unit,
+    fieldStateEmail: EmailFieldState,
+    modifier: Modifier = Modifier,
```
Comment (bstolyarov) [inline app/src/main/kotlin/com/softteco/template/ui/components/EmailField.kt:19]: image

**Review by EvgeniyFilippov — APPROVED** [review]

**Pull request merged**